### PR TITLE
Json type check

### PR DESCRIPTION
**Pull request by Mike-3**

### CODE REVIEW

**Comment by S-Man42** [comment]
Missing type checks for JSON Arrays and JSON Objects

File: lib/utils/common_utils.dart [hunk 1]
```diff
@@ -404,3 +404,40 @@ String normalizeCharacters(String input) {
 
   return input;
 }
+
+double jsonDouble(dynamic value) {
```
Comment (S-Man42) [inline lib/utils/common_utils.dart:408]: missing tests

File: lib/utils/common_utils.dart [hunk 2]
```diff
@@ -404,3 +404,40 @@ String normalizeCharacters(String input) {
...
+}
+
+bool jsonBool(dynamic value) {
+  if (value is bool)
+    return value;
+
+  return null;
+}
+
+String jsonString(dynamic value) {
```
Comment (S-Man42) [inline lib/utils/common_utils.dart:433]: I believe that's not working. Here you'd get something like Object is not Type String. I guess .toString() is necessary at least, but in my opinion a proper type check is necessary as well...

File: lib/utils/common_utils.dart [hunk 3]
```diff
@@ -404,3 +404,40 @@ String normalizeCharacters(String input) {
...
+    return value;
+}
+
+List<String> jsonStringList(List<dynamic> list) {
```
Comment (S-Man42) [inline lib/utils/common_utils.dart:437]: what about int arrays and so on?

File: lib/utils/common_utils.dart [hunk 4]
```diff
@@ -404,3 +404,40 @@ String normalizeCharacters(String input) {
...
+double jsonDouble(dynamic value) {
+  if (value is double)
+    return value;
+  else if (value is int)
+    return (value as int).toDouble();
+
+  return null;
+}
+
+int jsonInt(dynamic value) {
```
Comment (S-Man42) [inline lib/utils/common_utils.dart:417]: I'd see these functions in a separate json_utils

File: lib/utils/common_utils.dart [hunk 5]
```diff
@@ -404,3 +404,40 @@ String normalizeCharacters(String input) {
...
+  if (value is int)
+    return value;
+  else if (value is double)
```
Comment (S-Man42) [inline lib/utils/common_utils.dart:420]: Hm... discussion here... Getting a 1.2 as input value will be handled as 1? In my opinion it's an errornous input and should be handled as this. Maybe the using tool can decide what to do in error case, but the util function never should. Or we add an jsonDoubleOrInt method separately...

Comment (Mike-3) [reply lib/utils/common_utils.dart:420]: I just really didn't want to consider inputs from 1.0 as bugs.

File: lib/utils/common_utils.dart (hunk 3 above)
Comment (Mike-3) [reply lib/utils/common_utils.dart:437]: Wasn't necessary at the moment

File: lib/utils/common_utils.dart (hunk 4 above)
Comment (Mike-3) [reply lib/utils/common_utils.dart:417]: ok

**Comment by Mike-3** [comment]
rebase conflict error

**Pull request closed without merging**